From ec8c12a184fa571406e9d21db04859f6b7230991 Mon Sep 17 00:00:00 2001
From: Zhang Jiejing <jiejing.zhang@freescale.com>
Date: Tue, 29 May 2012 18:04:40 +0800
Subject: [PATCH 281/635] ENGR00210918 hardware: imx: add build flag to
 gralloc, hwcomposer.

add build flag to choose build gralloc, hwcomposer, libcamera lib.
for some chip(6SL) don't have GPU, so it pefer the use orignally no
hardware accelerated HAL, add flag to archive this.

Signed-off-by Zhang Jiejing <jiejing.zhang@freescale.com>
---
 hwcomposer/Android.mk             | 1 +
 mx5x/hwcomposer/Android.mk        | 2 ++
 mx5x/libcamera/Android.mk         | 3 ++-
 mx5x/libgralloc/Android.mk        | 5 ++++-
 mx6/hwcomposer/Android.mk         | 2 ++
 mx6/libcamera/Android.mk          | 4 ++++
 mx6/libgralloc_wrapper/Android.mk | 5 ++++-
 7 files changed, 19 insertions(+), 3 deletions(-)

diff --git a/hwcomposer/Android.mk b/hwcomposer/Android.mk
index ba6318b..ae5b6f3 100644
--- a/hwcomposer/Android.mk
+++ b/hwcomposer/Android.mk
@@ -26,3 +26,4 @@ LOCAL_MODULE := hwcomposer.$(TARGET_BOARD_PLATFORM)
 LOCAL_CFLAGS:= -DLOG_TAG=\"hwcomposer\"
 LOCAL_MODULE_TAGS := eng
 include $(BUILD_SHARED_LIBRARY)
+
diff --git a/mx5x/hwcomposer/Android.mk b/mx5x/hwcomposer/Android.mk
index 8c3211f..824b1cd 100755
--- a/mx5x/hwcomposer/Android.mk
+++ b/mx5x/hwcomposer/Android.mk
@@ -17,6 +17,7 @@ LOCAL_PATH := $(call my-dir)
 
 # HAL module implemenation, not prelinked and stored in
 # hw/<OVERLAY_HARDWARE_MODULE_ID>.<ro.product.board>.so
+ifeq ($(TARGET_HAVE_IMX_HWCOMPOSER),true)
 include $(CLEAR_VARS)
 LOCAL_PRELINK_MODULE := false
 LOCAL_MODULE_PATH := $(TARGET_OUT_SHARED_LIBRARIES)/hw
@@ -28,3 +29,4 @@ LOCAL_C_INCLUDES += external/linux-lib/ipu
 LOCAL_CFLAGS:= -DLOG_TAG=\"hwcomposer\"
 LOCAL_MODULE_TAGS := eng
 include $(BUILD_SHARED_LIBRARY)
+endif
diff --git a/mx5x/libcamera/Android.mk b/mx5x/libcamera/Android.mk
index 42c958b..0e6e3e3 100755
--- a/mx5x/libcamera/Android.mk
+++ b/mx5x/libcamera/Android.mk
@@ -14,7 +14,7 @@
 
 ifeq ($(BOARD_SOC_CLASS),IMX5X)
 LOCAL_PATH:= $(call my-dir)
-
+ifeq ($(BOARD_HAVE_IMX_CAMERA),true)
 include $(CLEAR_VARS)
 
 LOCAL_SRC_FILES:=    \
@@ -75,3 +75,4 @@ LOCAL_MODULE_TAGS := eng
 
 include $(BUILD_SHARED_LIBRARY)
 endif
+endif
diff --git a/mx5x/libgralloc/Android.mk b/mx5x/libgralloc/Android.mk
index 9fd0774..6a4d489 100755
--- a/mx5x/libgralloc/Android.mk
+++ b/mx5x/libgralloc/Android.mk
@@ -16,6 +16,8 @@ LOCAL_PATH := $(call my-dir)
 
 # HAL module implemenation, not prelinked and stored in
 # hw/<OVERLAY_HARDWARE_MODULE_ID>.<ro.product.board>.so
+
+ifeq ($(TARGET_HAVE_IMX_GRALLOC), true)
 include $(CLEAR_VARS)
 LOCAL_PRELINK_MODULE := true
 LOCAL_MODULE_PATH := $(TARGET_OUT_SHARED_LIBRARIES)/hw
@@ -33,7 +35,6 @@ LOCAL_SRC_FILES := 	\
 	gralloc.cpp 	\
 	framebuffer.cpp \
 	mapper.cpp
-	
 LOCAL_MODULE := gralloc.$(TARGET_BOARD_PLATFORM)
 LOCAL_CFLAGS:= -DLOG_TAG=\"$(TARGET_BOARD_PLATFORM).gralloc\" -D_LINUX
 
@@ -49,3 +50,5 @@ endif
 LOCAL_MODULE_TAGS := eng
 
 include $(BUILD_SHARED_LIBRARY)
+
+endif
diff --git a/mx6/hwcomposer/Android.mk b/mx6/hwcomposer/Android.mk
index 2c3d580..5d0e3ba 100755
--- a/mx6/hwcomposer/Android.mk
+++ b/mx6/hwcomposer/Android.mk
@@ -17,6 +17,7 @@ LOCAL_PATH := $(call my-dir)
 
 # HAL module implemenation, not prelinked and stored in
 # hw/<OVERLAY_HARDWARE_MODULE_ID>.<ro.product.board>.so
+ifeq ($(TARGET_HAVE_IMX_HWCOMPOSER),true)
 include $(CLEAR_VARS)
 LOCAL_PRELINK_MODULE := false
 LOCAL_MODULE_PATH := $(TARGET_OUT_SHARED_LIBRARIES)/hw
@@ -28,3 +29,4 @@ LOCAL_C_INCLUDES += external/linux-lib/ipu
 LOCAL_CFLAGS:= -DLOG_TAG=\"hwcomposer\"
 LOCAL_MODULE_TAGS := eng
 include $(BUILD_SHARED_LIBRARY)
+endif
diff --git a/mx6/libcamera/Android.mk b/mx6/libcamera/Android.mk
index 7d09df1..cea4761 100755
--- a/mx6/libcamera/Android.mk
+++ b/mx6/libcamera/Android.mk
@@ -15,6 +15,8 @@
 ifeq ($(BOARD_SOC_CLASS),IMX6)
 LOCAL_PATH:= $(call my-dir)
 
+ifeq ($(BOARD_HAVE_IMX_CAMERA),true)
+
 include $(CLEAR_VARS)
 
 LOCAL_SRC_FILES:=    \
@@ -74,3 +76,5 @@ LOCAL_MODULE_TAGS := eng
 
 include $(BUILD_SHARED_LIBRARY)
 endif
+
+endif
diff --git a/mx6/libgralloc_wrapper/Android.mk b/mx6/libgralloc_wrapper/Android.mk
index 47a558f..08c8e7a 100755
--- a/mx6/libgralloc_wrapper/Android.mk
+++ b/mx6/libgralloc_wrapper/Android.mk
@@ -16,6 +16,8 @@ LOCAL_PATH := $(call my-dir)
 
 # HAL module implemenation, not prelinked and stored in
 # hw/<OVERLAY_HARDWARE_MODULE_ID>.<ro.product.board>.so
+
+ifeq ($(TARGET_HAVE_IMX_GRALLOC),true)
 include $(CLEAR_VARS)
 LOCAL_PRELINK_MODULE := false
 LOCAL_MODULE_PATH := $(TARGET_OUT_SHARED_LIBRARIES)/hw
@@ -26,7 +28,7 @@ LOCAL_SRC_FILES := 	\
 	framebuffer.cpp \
 	mapper.cpp      \
         display_mode.cpp
-	
+
 LOCAL_MODULE := gralloc.$(TARGET_BOARD_PLATFORM)
 LOCAL_CFLAGS:= -DLOG_TAG=\"$(TARGET_BOARD_PLATFORM).gralloc\" -D_LINUX
 
@@ -42,3 +44,4 @@ endif
 LOCAL_MODULE_TAGS := eng
 
 include $(BUILD_SHARED_LIBRARY)
+endif
-- 
1.8.0

